### PR TITLE
UpdateLights Task

### PR DESCRIPTION
**Pull request by babusid**
Addresses Issue #109.  Implements the Update lights thread with the semaphore signaling the BlinkLights thread architecture. This may have to be revisited if we decide to remove the signaling semaphore for BlinkLights as discussed [here][https://github.com/lhr-solar/Controls/pull/168#discussion_r776517042](https://github.com/lhr-solar/Controls/pull/168/files#r762465782)

### CODE REVIEW

File: Apps/Src/UpdateLights.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            Lights_Set(A_CNCTR,ON);
+        } else {
+            Lights_Set(A_CNCTR,OFF);
+        }
+
+        if(GlobalCarState->ShouldMotorBeActivated==ON){
+            Lights_Set(M_CNCTR,ON);
+        } else {
+            Lights_Set(M_CNCTR,OFF);
+        }
```
Comment (ClarkPoon) [inline Apps/Src/UpdateLights.c:59]:
```suggestion
        Lights_Set(M_CNCTR, GlobalCarState->ShouldMotorBeActivated);
```

File: Apps/Src/UpdateLights.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            Lights_Set(Headlight_ON,ON);
+        } else {
+            Lights_Set(Headlight_ON,OFF);
+        }
+
+        if(GlobalLightSwitches->HZD_SW == ON || GlobalLightSwitches->LEFT_SW == ON || GlobalLightSwitches->RIGHT_SW ==ON){
+            OSSemPost(
+                &BlinkLight_Sem4,
+                OS_OPT_POST_ALL,
+                &err
```
Comment (ClarkPoon) [inline Apps/Src/UpdateLights.c:36]: Where is this semaphore initialized? 

Also, I don't think this is the best way to do this. Why not just have the blink lights task run constantly at 1.5 Hz and read `GlobalLightSwitches` directly?

**Review by ClarkPoon — COMMENTED** [review]
Why did you create `Apps/Src/BlinkLights.c` as an empty file?

**Comment by babusid** [comment]
This task needs to be turned into functional logic that is integrated into the ReadSwitches task. This task will then be deprecated and deleted after ReadSwitches verification.

**Pull request closed without merging**